### PR TITLE
fix entity parenting crash

### PR DESCRIPTION
**Pull request by druiz17**
This PR fixes the issue of the ```_needsParentFixupLock``` being clocked twice when ```EntityTree::fixupNeedsParentFixup``` function is called.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6672/app_12279_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6672/HighFidelity-Beta-PR12279-68fd9eafa8c3fd7409d410750be1cae162ec89b1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6672/HighFidelity-Beta-PR12279-68fd9eafa8c3fd7409d410750be1cae162ec89b1.exe

**Comment by druiz17** [comment]
Test plan (two or more testers needed)
- run the script below
- the script should add a cube and a sphere at the same location and also add a ```Parent``` button to toolbar or tablet.
- click the ```parent``` button should parent the entities together causing them to disappear.
- make sure that other user viewing the parenting don't crash 
- after around 9 seconds the objects should reappear and repeat the process
- have other testers run the script as well.

File: libraries/entities/src/EntityTree.cpp [hunk 1]
```diff
@@ -1748,6 +1757,8 @@ void EntityTree::fixupNeedsParentFixups() {
         PerformanceTimer perfTimer("recurseTreeWithOperator");
         recurseTreeWithOperator(&moveOperator);
     }
+
+    setNeedsParentFixup(entitiesParentFixup);
```
Comment (sethalves) [inline libraries/entities/src/EntityTree.cpp]: I'm worried about this... is it possible for entities to be added to _needsParentFixup in between the calls to getEntitiesParentFixup() and setNeedsParentFixup()?

Comment (druiz17) [reply libraries/entities/src/EntityTree.cpp]: There is actually a writelock on _needsParentFixup that is calling in the middle of this function.

Comment (druiz17) [reply libraries/entities/src/EntityTree.cpp]: I'll think about a better solution.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6709/app_12279_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6709/HighFidelity-Beta-PR12279-bed2ea052df8dbe4f08b501dd0646b0eeb905d7a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6709/HighFidelity-Beta-PR12279-bed2ea052df8dbe4f08b501dd0646b0eeb905d7a.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6710/app_12279_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6710/HighFidelity-Beta-PR12279-4c0a1732874c205d3235f1ae28536c8b648f31d7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6710/HighFidelity-Beta-PR12279-4c0a1732874c205d3235f1ae28536c8b648f31d7.exe

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by sbennett77** [comment]
Passed.

@conklin94122

**Pull request merged**